### PR TITLE
Set up saving of EntryBook to XML; standardized EntryBook/Storage classes with AB4

### PR DESCRIPTION
**Pull request by marvintxd**
- Added infrastructure to save EntryBook to an XML file.
- Made `StorageManager` implement and contain `EntryBookStorage`, as it did with `AddressBookStorage`.
- Made `MainApp` load `EntryBook` through `Storage` component rather than using hardcoded filepath.

TODO: Link Model -> Events -> Storage -- create events for when `EntryBook` changes in the Model.

### CODE REVIEW

File: src/main/java/seedu/address/storage/entry/XmlSerializableEntryBook.java [hunk 1]
```diff
@@ -29,6 +31,13 @@ public XmlSerializableEntryBook() {
         resumeEntries = new ArrayList<>();
     }
 
+    /**
+     * Conversion
+     */
+    public XmlSerializableEntryBook(ReadOnlyEntryBook src) {
+        this();
+        resumeEntries.addAll(src.getEntryList().stream().map(XmlAdaptedResumeEntry::new).collect(Collectors.toList()));
```
Comment (anubh-v) [inline src/main/java/seedu/address/storage/entry/XmlSerializableEntryBook.java]: My suggestion is to break up this part into a few statements for more readability.

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 2]
```diff
@@ -91,6 +96,35 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) thro
         addressBookStorage.saveAddressBook(addressBook, filePath);
     }
 
+    // ================ EntryBook methods ==============================
+
+    @Override
+    public Path getEntryBookFilePath() {
+        return addressBookStorage.getAddressBookFilePath();
```
Comment (anubh-v) [inline src/main/java/seedu/address/storage/StorageManager.java]: Typo here?

**Comment by anubh-v** [comment]
Looks ok:
We need to add tests for XmlEntryBookStorage.

I have updated the issue: #264

File: src/main/java/seedu/address/storage/StorageManager.java (hunk 2 above)
Comment (marvintxd) [reply src/main/java/seedu/address/storage/StorageManager.java]: Thanks, changed.

File: src/test/java/seedu/address/storage/entry/XmlEntryBookStorageTest.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    @Rule
+    public TemporaryFolder testFolder = new TemporaryFolder();
+
+    @Test
+    public void readEntryBook_nullFilePath_throwsNullPointerException() throws Exception {
+        thrown.expect(NullPointerException.class);
+        readEntryBook(null);
+    }
+
+    private java.util.Optional<ReadOnlyEntryBook> readEntryBook(String filePath) throws Exception {
```
Comment (anubh-v) [inline src/test/java/seedu/address/storage/entry/XmlEntryBookStorageTest.java]: typo here? You can import Optional

Comment (marvintxd) [reply src/test/java/seedu/address/storage/entry/XmlEntryBookStorageTest.java]: It was like that in the AB storage test for some reason haha. But can change too

**Pull request merged**